### PR TITLE
Issue #2215 Add note in docs about the limitations of shared pool

### PR DESCRIPTION
**Pull request by rahul-mittal**

### CODE REVIEW

File: docs/src/docs/asciidoc/user/clustered-cache.adoc [hunk 1]
```diff
@@ -143,6 +143,11 @@ In this example, a dedicated pool of 32MB is allocated for clustered-cache from
 <7> Configures another cache (`shared-cache-2`) that shares the resource pool (`resource-pool-a`) with `shared-cache-1`.
 <8> Creates fully initialized cache manager with the clustered caches.
 
+NOTE:  When multiple caches are sharing a shared pool, eviction operation called by a cache evicts elements of that cache
+itself and not just any set of elements in the shared pool. This means that today, when space in a shared pool
+is allocated to a cache, it remains with it for its use only. Improvisation would be made in future to allow eviction of
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/clustered-cache.adoc]: Improvisation?!?

File: docs/src/docs/asciidoc/user/clustered-cache.adoc [hunk 2]
```diff
@@ -143,6 +143,11 @@ In this example, a dedicated pool of 32MB is allocated for clustered-cache from
 <7> Configures another cache (`shared-cache-2`) that shares the resource pool (`resource-pool-a`) with `shared-cache-1`.
 <8> Creates fully initialized cache manager with the clustered caches.
 
+NOTE:  When multiple caches are sharing a shared pool, eviction operation called by a cache evicts elements of that cache
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/clustered-cache.adoc]: evictions

File: docs/src/docs/asciidoc/user/clustered-cache.adoc (hunk 1 above)
Comment (henri-tremblay) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: And we should not talk about the future. This is the current documentation

File: docs/src/docs/asciidoc/user/clustered-cache.adoc (hunk 2 above)
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/clustered-cache.adoc]: Am I right to say that the real thing is that a cache will reserve a block of memory from the shared pool and keep it forever? So eviction or not, this allocated memory will stay with a given cache and that it's. If I am right, that's probably what you should explain.

File: docs/src/docs/asciidoc/user/clustered-cache.adoc (hunk 1 above)
Comment (rahul-mittal) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: removed

File: docs/src/docs/asciidoc/user/clustered-cache.adoc (hunk 2 above)
Comment (rahul-mittal) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: nothing on eviction now in the new version

Comment (rahul-mittal) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: revised it, please let me know if it sounds better now

File: docs/src/docs/asciidoc/user/clustered-cache.adoc [hunk 3]
```diff
@@ -143,6 +143,9 @@ In this example, a dedicated pool of 32MB is allocated for clustered-cache from
 <7> Configures another cache (`shared-cache-2`) that shares the resource pool (`resource-pool-a`) with `shared-cache-1`.
 <8> Creates fully initialized cache manager with the clustered caches.
 
+NOTE:  When a cache is allocated a block of memory from a shared pool, it is retained forever and would never get
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/clustered-cache.adoc]: I would put NOTE in normal case but in bold **Note:**. And only one space after

Comment (rahul-mittal) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: 'NOTE' here is asciidoc syntax for admonition para, 'Note' will not have the same effect. Removed one extra space though.

Comment (henri-tremblay) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: Cool. Didn't know that.

File: docs/src/docs/asciidoc/user/clustered-cache.adoc [hunk 4]
```diff
@@ -143,6 +143,9 @@ In this example, a dedicated pool of 32MB is allocated for clustered-cache from
 <7> Configures another cache (`shared-cache-2`) that shares the resource pool (`resource-pool-a`) with `shared-cache-1`.
 <8> Creates fully initialized cache manager with the clustered caches.
 
+NOTE: When a cache is allocated a block of memory from a shared pool, it is retained forever and would never get
```
[the remark this answers is not in the document]
Comment (henri-tremblay) [reply docs/src/docs/asciidoc/user/clustered-cache.adoc]: I'm not fully happy with it either. I'll call it a good start. CAUTION seems a good idea. If you have better explanation of the impacts, please do.

**Pull request merged**